### PR TITLE
CoW warning mode: setting values into single column of DataFrame

### PR DESCRIPTION
**Pull request by jorisvandenbossche**
Next subtask of https://github.com/pandas-dev/pandas/issues/56019. 

This tackles `mgr.column_setitem`, which is used to set values into a single column (but not setting the full column, so like `df.loc[idx, "col"] = ..`)

### CODE REVIEW

**Comment by jorisvandenbossche** [comment]
Green!

File: pandas/core/internals/blocks.py [hunk 1]
```diff
@@ -1737,13 +1737,14 @@ def round(self, decimals: int, using_cow: bool = False) -> Self:
...
         if values is self.values:
-            refs = self.refs
             if not using_cow:
                 # Normally would need to do this before, but
                 # numpy only returns same array when round operation
                 # is no-op
                 # https://github.com/numpy/numpy/blob/486878b37fc7439a3b2b87747f50db9b62fea8eb/numpy/core/src/multiarray/calculation.c#L625-L636
                 values = values.copy()
+            else:
+                refs = self.refs
```
Comment (jorisvandenbossche) [inline pandas/core/internals/blocks.py:1747]: We were incorrectly adding forwarding refs always, even when we did make a copy. For CoW this doesn't cause anything wrong, but for the warning mode this gave false positive warnings.

Comment (phofl) [reply pandas/core/internals/blocks.py:1747]: Yeah sorry for that, I didn't care much about non CoW when adding all of those

File: pandas/tests/copy_view/test_core_functionalities.py [hunk 2]
```diff
@@ -28,27 +28,32 @@ def test_setitem_dont_track_unnecessary_references(using_copy_on_write):
...
+    using_copy_on_write, warn_copy_on_write, request
+):
     df = DataFrame({"a": [1, 2, 3], "b": 1, "c": 1})
     view = df[:]
 
     df["b"] = 100
     arr = get_array(df, "a")
     view = None  # noqa: F841
-    df.iloc[0, 0] = 100
+    # TODO(CoW-warn) false positive?
```
Comment (phofl) [inline pandas/tests/copy_view/test_core_functionalities.py:54]: Yeah, comment explaining this is 3 lines below

**Review by phofl — APPROVED** [review]

**Comment by phofl** [comment]
thx @jorisvandenbossche

**Pull request merged**